content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,423 @@
+# Checkpoints
+
+ML checkpoints are an important part of deep learning because ML engineers like
+to save the model files at certain points during a training process.
+
+With DVC experiments and checkpoints, you can:
+
+- Implement the best practice in deep learning to save your model weights as
+  checkpoints.
+- Add reproducibility to the messy deep learning process.
+- Track all code and data changes corresponded to the checkpoints.
+- Return a report when metrics start diverging.
+
+You can also automate the process of tracking every training epoch, return to
+any previous checkpoint, see when metrics converge or when the learning rate
+needs to be adjusted, and even recover model weights, parameters, and code. When
+you adjust parameters and code, DVC tracks those changes for you without adding
+a lot of commits to your workspace.
+
+[The way checkpoints are implemented by DVC](/blog/experiment-refs) utilizes
+_ephemeral_ experiment commits and experiment branches within DVC. They are
+created using the metadata from experiments and are tracked with the `exps`
+custom Git reference.
+
+You can add experiments to your Git history by committing the experiment you
+want to track, which you'll see later in this tutorial.
+
+## Setting up the project
+
+This tutorial is going to cover how to implement checkpoints in a ML project
+using DVC. We're going to train a model to identify handwritten digits based on
+the MNIST dataset.
+
+You can follow along with the steps here or you can clone the repo directly from
+GitHub and play with it. To clone the repo, run the following commands.
+
+```bash
+git clone https://github.com/iterative/checkpoints-tutorial
+cd checkpoints-tutorial
+```
+
+It is highly recommended you create a virtual environment for this example. You
+can do that by running:
+
+```bash
+python3 -m venv .venv
+```
+
+Once your virtual environment is installed, you can start it up with one of the
+following commands.
+
+- On Mac/Linux: `source .venv/bin/activate`
+- On Windows: `.\.venv\Scripts\activate`
+
+Once you have your environment set up, you can install the dependencies by
+running:
+
+```bash
+pip install -r requirements.txt
+```
+
+This will download all of the packages you need to run the example. Now you have
+everything you need to get started with experiments and checkpoints.
+
+## Setting up a DVC pipeline
+
+If you don't have a DVC pipeline setup in your project, adding one only takes a
+few commands. At the root of the project, run:
```

Comment:
Some motivation for DVC pipeline would be helpful. Like `DVC versions data and models and it also can version the ML model weights file (the checkpoints) during the training. To enable this you will need to run your model training under DVC pipeline. Adding DVC pipline only takes a few commands.`